### PR TITLE
fix: update soilweb code for new API (color, edit URL, distance)

### PR DESCRIPTION
**Pull request by paulschreiber**
## Description
update soilweb code for new API (color, edit URL, distance)

### CODE REVIEW

**Comment by jjmaynard** [comment]
@paulschreiber branch ready to be merged to main

File: soil_id/us_soil.py [hunk 1]
```diff
@@ -373,6 +374,15 @@ def list_soils(lon, lat):
...
         comppct = pd.DataFrame([group_sorted.comppct_r.unique()] * len(sand_pct_intpl))
         cokey = pd.DataFrame([group_sorted.cokey.unique()] * len(sand_pct_intpl))
+
+        # Print shapes of DataFrames to debug
+        print(sand_pct_intpl[["c_sandpct_intpl_grp"]].shape)
+        print(clay_pct_intpl[["c_claypct_intpl_grp"]].shape)
+        print(cf_pct_intpl[["c_cfpct_intpl_grp"]].shape)
+        print(compname.shape)
+        print(cokey.shape)
+        print(comppct.shape)
```
Comment (paulschreiber) [inline soil_id/us_soil.py:384]:
```suggestion
```

File: soil_id/us_soil.py [hunk 2]
```diff
@@ -537,11 +550,27 @@ def list_soils(lon, lat):
...
+            # OSDhorzdata_pd = OSDhorzdata_pd.rename(
+            #     columns={
+            #         "srgb_r": "r",
+            #         "srgb_g": "g",
+            #         "srgb_b": "b",
+            #         "cielab_l": "l",
+            #         "cielab_a": "a",
+            #         "cielab_b": "b",
+            #     }
+            # )
```
Comment (paulschreiber) [inline soil_id/us_soil.py:572]:
```suggestion
```

File: soil_id/utils.py [hunk 3]
```diff
@@ -2575,8 +2325,8 @@ def update_esd_data(df):
     Processes the given DataFrame by updating missing ESD data based on component
     groups with the same names, filling missing URLs and ecoclass IDs and names.
     """
-    if "esd_url" not in df.columns:
-        df["esd_url"] = np.nan  # Initialize 'esd_url' as NaN if it does not exist
+    if "edit_url" not in df.columns:
+        df["edit_url"] = np.nan  # Initialize 'edit_url' as NaN if it does not exist
```
Comment (paulschreiber) [inline soil_id/utils.py:2329]: question: should this be `None` or `""`?

Comment (jjmaynard) [reply soil_id/utils.py]: Yes, should be: df["edit_url"] = ""
I've updated code

File: soil_id/us_soil.py (hunk 2 above)
Comment (jjmaynard) [reply soil_id/us_soil.py:572]: Yes, remove

**Pull request merged**